nengo/builder/tests/test_learning_rules.py

```diff
@@ -10,6 +16,50 @@ def test_simpes():
     assert str(sim) == "SimPES{pre=pre_filtered, error=error -> delta}"
 
 
+@pytest.mark.parametrize("pre_neurons", (True, False))
+@pytest.mark.parametrize("post_neurons", (True, False))
+@pytest.mark.parametrize("weight_solver", (True, False))
+def test_build_pes(pre_neurons, post_neurons, weight_solver):
```

Comment:
Since this test is fast, maybe it would be good to test slices on pre/post too? The one downside to a builder test like this is it won't apply to backends.

Reply:
Good call on checking slices, because it turns out the PES rule didn't work at all with a slice on `pre` 😂. I also modified it to use a Simulator (it should still be pretty fast, because we're only running it for one step), since NengoOCL and NengoDL both have their own PES builders which we'd want to check for these cases as well.